scalding-commons/src/main/scala/com/twitter/scalding/commons/thrift/TBaseOrderedSerialization.scala

```diff
@@ -3,28 +3,57 @@ package com.twitter.scalding.commons.thrift
 import com.twitter.scalding.serialization.{ OrderedSerialization, Serialization }
 import cascading.tuple.hadoop.io.BufferedInputStream
 import cascading.tuple.StreamComparator
-import org.apache.thrift.protocol.TCompactProtocol
-import org.apache.thrift.protocol.TBinaryProtocol
+import org.apache.thrift.protocol._
 import org.apache.thrift.TBase
 import java.io.{ ByteArrayOutputStream, OutputStream, InputStream }
 import org.apache.thrift.transport.TIOStreamTransport
+import scala.reflect.ClassTag
 
 import scala.util.{ Failure, Success, Try }
 import scala.util.control.NonFatal
 
 import com.twitter.scalding.serialization.JavaStreamEnrichments._
 
-abstract class TBaseOrderedBinary[T <: TBase[_, _]] extends TProtocolOrderedSerialization[T] {
+object TBaseOrderedSerialization {
+  implicit def defaultTBaseOrderedSerialization[T <: TBase[_, _]: ClassTag] = new TBaseOrderedSerialization {
+    override val bufferSize = 512
+  }
+}
+
+abstract class TBaseOrderedSerialization[T <: TBase[_, _]: ClassTag] extends TProtocolOrderedSerialization[T] {
```

Comment:
might be better to just take the Class[T] here than the ClassTag, since that's all we need, right? Otherwise, we are going to have to serialize the ClassTag. I'd be more confident we are serializing the Class correctly.